v2/pkg/engine/resolve/response.go

```diff
@@ -23,13 +23,20 @@ type GraphQLSubscriptionTrigger struct {
 	QueryPlan      *QueryPlan
 }
 
+// GraphQLResponse contains intermediate and final data of the response.
+// Fields are filled in this order:
+//
+//  1. Planner fills RawFetches and Info fields.
+//  2. PostProcessor fills Fetches and DataSources fields.
+//  3. Resolver fills the Data.
```

Comment:
3. Loader executes Fetches and collects all json data
4. Resolver uses Data to create a final json shape to return as response